### PR TITLE
Exit Run() method when receiving the Stop signal

### PR DESCRIPTION
**Pull request by cyrilleverrier**

### CODE REVIEW

**Comment by cyrilleverrier** [comment]
@ruflin : when I run the system-test with this fix, I got this data race:
Is it related to the new code in libbeat?

```
2016/01/25 19:33:45.738945 beat.go:220: DBG  Initializing output plugins
2016/01/25 19:33:45.739002 geolite.go:24: INFO GeoIP disabled: No paths were set under output.geoip.paths
2016/01/25 19:33:45.739033 file.go:39: INFO File output base filename set to: examplebeat
2016/01/25 19:33:45.739064 file.go:50: INFO Rotate every bytes set to: 10485760
2016/01/25 19:33:45.739088 file.go:57: INFO Number of files set to: 7
2016/01/25 19:33:45.739180 outputs.go:119: INFO Activated file as output plugin.
2016/01/25 19:33:45.739210 publish.go:232: DBG  Create output worker
2016/01/25 19:33:45.739375 publish.go:274: DBG  No output is defined to store the topology. The server fields might not be filled.
2016/01/25 19:33:45.739420 publish.go:288: INFO Publisher name: dell
2016/01/25 19:33:45.739682 async.go:78: INFO Flush Interval set to: -1ms
2016/01/25 19:33:45.739710 async.go:84: INFO Max Bulk Size set to: -1
2016/01/25 19:33:45.739742 beat.go:230: INFO Init Beat: examplebeat; Version: 1.2.0-SNAPSHOT
2016/01/25 19:33:45.741799 beat.go:259: INFO examplebeat sucessfully setup. Start running.
examplebeat is running! Hit CTRL-C to stop it.
2016/01/25 19:33:45.829638 service.go:30: DBG  Received sigterm/sigint, stopping
2016/01/25 19:33:45.829720 beat.go:299: INFO Start exiting beat
2016/01/25 19:33:45.829764 beat.go:274: INFO Stopping Beat
==================
WARNING: DATA RACE
Read by goroutine 8:
  github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat.(*Beat).Stop()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:276 +0x7b
  github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat.Run()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:137 +0xf0
  github.com/your-github-name/examplebeat.main()
      github.com/your-github-name/examplebeat/_test/_obj_test/main.go:12 +0x96
  github.com/your-github-name/examplebeat.TestSystem()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/main_test.go:20 +0x54
  testing.tRunner()
      /usr/lib/go/src/testing/testing.go:456 +0xdc

Previous write by goroutine 9:
  github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat.(*Beat).Run()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:261 +0x40f
  github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat.(*Beat).Start()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:173 +0x15e
  github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat.Run.func1()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:120 +0x4c

Goroutine 8 (running) created at:
  testing.RunTests()
      /usr/lib/go/src/testing/testing.go:561 +0xaa3
  testing.(*M).Run()
      /usr/lib/go/src/testing/testing.go:494 +0xe4
  main.main()
      github.com/your-github-name/examplebeat/_test/_testmain.go:112 +0x38d

Goroutine 9 (running) created at:
  github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat.Run()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:132 +0x9a
  github.com/your-github-name/examplebeat.main()
      github.com/your-github-name/examplebeat/_test/_obj_test/main.go:12 +0x96
  github.com/your-github-name/examplebeat.TestSystem()
      /home/cyrille/code/go/workspace/src/github.com/your-github-name/examplebeat/main_test.go:20 +0x54
  testing.tRunner()
      /usr/lib/go/src/testing/testing.go:456 +0xdc
==================
2016/01/25 19:33:45.830244 beat.go:282: INFO Cleaning up examplebeat before shutting down.
2016/01/25 19:33:45.830401 beat.go:138: INFO Exit beat completed
PASS
coverage: 92.3% of statements in ./beater/..., ./config/..., .
Found 1 data race(s)
```

File: {{cookiecutter.beat}}/beater/{{cookiecutter.beat}}.go [hunk 1]
```diff
@@ -41,6 +40,8 @@ func (bt *{{cookiecutter.beat|capitalize}}) Setup(b *beat.Beat) error {
 
 func (bt *{{cookiecutter.beat|capitalize}}) Run(b *beat.Beat) error {
 	fmt.Println("{{cookiecutter.beat}} is running! Hit CTRL-C to stop it.")
+	
+	<-bt.done
```
Comment (ruflin) [inline {{cookiecutter.beat}}/beater/{{cookiecutter.beat}}.go]: This is actually also done here: https://github.com/elastic/beat-generator/pull/2

**Comment by ruflin** [comment]
@cyrilleverrier It somehow looks like it, even though the lines reported could not be correct. I would not expect a race condition as there is only 1 instance of the beat.

**Comment by ruflin** [comment]
@cyrilleverrier I would really like to get travis builds into the game as soon as possible so we directly detect all these issues. Somehow travis is having issues at the moment and doesn't start the builds :-(

**Comment by ruflin** [comment]
@cyrilleverrier merged anyway as the commits look good.

**Pull request merged**